rosette/src/CommandLine.cc

```diff
@@ -71,114 +97,117 @@ int ParseCommandLine(int argc, char** argv) {
      * BigBang.cc for details.
      */
 
-    const char* cmd_name = argv[0];
+    int verbose_flag = 0;
+    int enable_repl = 0;
+    int c = 0;
+
+    const struct option long_options[] = {
+        /* Flags */
+        {"verbose", no_argument, &verbose_flag, 1},
+        {"quiet", no_argument, &verbose_flag, 0},
+        {"interactive-repl", no_argument, &enable_repl, 'i'},
+
+        /* Non-flags */
+        {"help", no_argument, 0, 'h'},
+        {"tenure", required_argument, 0, 't'},
+        {"paranoid-gc", required_argument, 0, 'p'},
+        {"infant-size", required_argument, 0, 'I'},
+        {"survivor-size", required_argument, 0, 's'},
+        {"old-size", required_argument, 0, 'o'},
+        {"boot-dir", required_argument, 0, 'd'},
+        {"boot", required_argument, 0, 'b'},
+        {0, 0, 0, 0},
+    };
 
     /*
      * We always manually reset ParanoidAboutGC in this routine so that
      * its value starts out at 0 even when we are actually restoring a
      * dumped image.
      */
-
     ParanoidAboutGC = 0;
 
-    DeleteArgs(0, argc, argv); /* Delete argv[0] (the command name) */
-    argc--;
-
-    int i = 0;
-
-    while (i < argc) {
-        if (strcmp(argv[i], "-t") == 0) {
-            if (!(i + 1 < argc && sscanf(argv[i + 1], "%d", &TenuringAge))) {
-                usage(cmd_name);
-            }
-
-            DeleteArgs(i, argc, argv, 2);
-            argc -= 2;
-            continue;
-        }
-
-        if (strcmp(argv[i], "-p") == 0) {
-            int temp = 0;
-            if (i + 1 < argc && sscanf(argv[i + 1], "%d", &temp)) {
-                ParanoidAboutGC = temp;
-                DeleteArgs(i, argc, argv, 2);
-                argc -= 2;
-            } else {
-                ParanoidAboutGC = 1;
-                DeleteArgs(i, argc, argv);
-                argc--;
-            }
-
-            continue;
-        }
-
-        if (strcmp(argv[i], "-is") == 0) {
-            int temp;
-            if (i + 1 < argc && sscanf(argv[i + 1], "%d", &temp)) {
-                InfantSpaceSize = temp * 1024;
-            } else {
-                usage(cmd_name);
-            }
-
-            DeleteArgs(i, argc, argv, 2);
-            argc -= 2;
-            continue;
-        }
+    while (1) {
+        int option_index = 0;
+        c = getopt_long(argc, argv, "+qt:p:i:s:o:b:",
```

Comment:
Little `i` takes no argument; big `I` missing, `h` missing, `v` missing, `d` missing.

Reply:
Thank you! Fix incoming...

Reply:
Done.